### DISCUSSION TITLE
xpack cmake in eclipse

### DISCUSSION THREAD

**Discussion by ksrp1984** [xpack-dev-tools/cmake-xpack#11] [Q&A]
Will there be support for Xpack-Сmake in Eclipse, like it is done in VS?
Or Embedded C/C++ Development Tool(Debuggers) for VS Code with file-api support? 
File-api is already supported by VS cmake-tools.

**Comment by ilg-ul** [comment]
xPack CMake is just a binary distribution of CMake.

What king of support do you expect?

**Reply by ksrp1984** [reply]
> What king of support do you expect?
Launching different configurations from the IDE. And building a project tree, as it is implemented in VS.
Eclipse has a cmake4eclipse plugin, but it works poorly. It also uses the compile_commands.json.
VS Code uses the Cmake file-api, which gives much more information about the project.
But I would prefer good support for debugging MCU (ARM, RISC-V) in VS, as it is done in Eclipse. 
I like VS better, but so far there is only a plugin for Cortex

**Comment by ksrp1984** [comment]


**Reply by ilg-ul** [reply]
> Launching different configurations from the IDE. 

For launching different configurations from VS Code I use [xPack C/C++ Managed Build Tools](https://marketplace.visualstudio.com/items?itemName=ilg-vscode.xpack), it works for any builder, including CMake and meson.

> Eclipse has a cmake4eclipse plugin

I'm not involved with this plugin, I suggest you ask them directly.

> But I would prefer good support for debugging MCU (ARM, RISC-V) in VS, as it is done in Eclipse.
> I like VS better, but so far there is only a plugin for Cortex

I prefer VS Code too, but for debug sessions I still use Eclipse.

I plan some future work on improving xPack support in VS Code, but it'll probably take some time to get there.

**Reply by ksrp1984** [reply]
> I plan some future work on improving xPack

The whole world will be grateful to you )))
